homeassistant/helpers/translation.py

```diff
@@ -528,6 +529,34 @@ def async_translations_loaded(hass: HomeAssistant, components: set[str]) -> bool
     )
 
 
+@callback
+def async_get_exception_message(
+    translation_domain: str | None,
+    translation_key: str | None,
+    translation_placeholders: dict[str, str] | None = None,
+    language: str = "en",
+) -> str | None:
+    """Get exception message from translation cache."""
+    if translation_domain is None or translation_key is None:
+        return None
+
+    hass = async_get_hass()
+    localize_key = (
+        f"component.{translation_domain}.exceptions.{translation_key}.message"
+    )
+    translations = async_get_cached_translations(hass, language, "exceptions")
+    if localize_key in translations:
+        if message := translations[localize_key]:
+            message = message.rstrip(".")
+        if not translation_placeholders:
+            return message
+        with suppress(KeyError):
+            message = message.format(**translation_placeholders)
+        return message
+
+    return None
```

Comment:
I think we should return the translation_key here, then at least the message is somewhat useful

Reply:
Returning `localize_key` when no translation was found in the cache.